aptos-move/framework/aptos-token-objects/sources/collection.move

```diff
@@ -157,22 +205,39 @@ module aptos_token_objects::collection {
         let constructor_ref = object::create_named_object(creator, collection_seed);
         let object_signer = object::generate_signer(&constructor_ref);
 
-        let supply = UnlimitedSupply {
-            current_supply: 0,
-            total_minted: 0,
-            burn_events: object::new_event_handle(&object_signer),
-            mint_events: object::new_event_handle(&object_signer),
-        };
-
-        create_collection_internal(
-            creator,
-            constructor_ref,
-            description,
-            name,
-            royalty,
-            uri,
-            option::some(supply),
-        )
+        if (features::concurrent_assets_enabled()) {
```

Comment:
Why not refactor this into a separate function for code reuse between this and create_fixed_collection?

Reply:
it's different - one calls create_unbounded_aggregator, while other aggregator_v2::create_aggregator(max_supply)